extension/chrome/elements/composer/composer-pwd-or-pubkey-container.ts

```diff
@@ -16,26 +18,33 @@ export class ComposerPwdOrPubkeyContainer extends ComposerComponent {
 
   public initActions = async () => {
     this.composer.S.cached('input_password').keyup(this.view.setHandlerPrevent('spree', () => this.showHideContainerAndColorSendBtn()));
-    this.composer.S.cached('input_password').focus(this.view.setHandlerPrevent('spree', () => {
+    this.composer.S.cached('input_password').focus(this.view.setHandlerPrevent('spree', async () => {
       const passwordContainerHeight = this.composer.S.cached('password_or_pubkey').outerHeight() || 0;
       const footerHeight = this.composer.S.cached('footer').outerHeight() || 0;
       this.composer.S.cached('expiration_note').css({
         bottom: (passwordContainerHeight + footerHeight) + 'px'
       });
       this.composer.S.cached('expiration_note').fadeIn();
-      this.showHideContainerAndColorSendBtn();
+      await this.showHideContainerAndColorSendBtn();
     }));
-    this.composer.S.cached('input_password').blur(() => {
-      this.composer.S.cached('expiration_note').fadeOut();
-      this.showHideContainerAndColorSendBtn();
+    this.composer.S.cached('input_password').blur(async () => {
+      setTimeout(() => { // timeout here is needed so <a> will be visible once clicked
+        this.composer.S.cached('expiration_note').fadeOut();
+      }, 100);
+      await this.showHideContainerAndColorSendBtn();
     });
+    this.composer.S.cached('expiration_note').find('#expiration_note_settings_link').click(this.view.setHandler((el, e) => {
+      e.preventDefault();
+      this.composer.render.renderSettingsWithDialog('security'), this.composer.errs.handlers(`render settings dialog`);
```

Comment:
I'm surprised this is valid syntax. The `this.composer.errs.handlers("render settings dialog")` should be a parameter passed into `setHandler`.

Reply:
Oops, fixed :+1: